src/widgets/index/__tests__/index-test.ts

```diff
@@ -2882,4 +2888,245 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/index-widge
+      );
+      expect(nestedIndex2.getResults()).toEqual(nestedIndex2ExpectedResults);
+    });
+
+    it('does not fail with non-provided index results', () => {
+      const search = new InstantSearch({
+        indexName: 'indexName',
+        searchClient: createSearchClient(),
+      });
+      search._initialResults = {
+        indexName: new SearchResults(new SearchParameters(), [
+          createSingleSearchResponse({
+            hits: [{ objectID: '1' }, { objectID: '2' }, { objectID: '3' }],
+          }),
+        ]),
+        // Notice that `indexName2` is not provided
+        indexName3: new SearchResults(new SearchParameters(), [
+          createSingleSearchResponse({
+            hits: [{ objectID: '7' }, { objectID: '8' }, { objectID: '9' }],
+          }),
+        ]),
+      };
+
+      const nestedIndex1 = index({ indexName: 'indexName2' });
+      const nestedIndex2 = index({ indexName: 'indexName3' });
+
+      nestedIndex1.addWidgets([nestedIndex2]);
+      search.addWidgets([nestedIndex1]);
+      search.start();
+
+      const indexExpectedResults = expect.objectContaining({
+        _rawResults: [
+          expect.objectContaining({
+            hits: [{ objectID: '1' }, { objectID: '2' }, { objectID: '3' }],
+          }),
+        ],
+        hits: [{ objectID: '1' }, { objectID: '2' }, { objectID: '3' }],
+      });
+      const nestedIndex1ExpectedResults = null;
+      const nestedIndex2ExpectedResults = expect.objectContaining({
+        _rawResults: [
+          expect.objectContaining({
+            hits: [{ objectID: '7' }, { objectID: '8' }, { objectID: '9' }],
+          }),
+        ],
+        hits: [{ objectID: '7' }, { objectID: '8' }, { objectID: '9' }],
+      });
+
+      expect(search.mainIndex.getHelper()!.lastResults).toEqual(
+        indexExpectedResults
+      );
+      expect(search.mainIndex.getResults()).toEqual(indexExpectedResults);
+      expect(nestedIndex1.getHelper()!.lastResults).toEqual(
+        nestedIndex1ExpectedResults
+      );
+      expect(nestedIndex1.getResults()).toEqual(nestedIndex1ExpectedResults);
+      expect(nestedIndex2.getHelper()!.lastResults).toEqual(
+        nestedIndex2ExpectedResults
+      );
+      expect(nestedIndex2.getResults()).toEqual(nestedIndex2ExpectedResults);
+    });
+
+    it('schedules a render after init', () => {
+      const instance = index({ indexName: 'indexName' });
+      const instantSearchInstance = createInstantSearch({
+        scheduleRender: jest.fn() as any,
+      });
+      instantSearchInstance._initialResults = {
+        indexName: new SearchResults(new SearchParameters(), [
+          createSingleSearchResponse({
+            hits: [{ objectID: '1' }, { objectID: '2' }, { objectID: '3' }],
+          }),
+        ]),
+      };
+
+      instance.init(createInitOptions({ instantSearchInstance, parent: null }));
+
+      expect(instantSearchInstance.scheduleRender).toHaveBeenCalledTimes(1);
```

Comment:
not sure if we need that test, but we could assert that scheduleRender isn't called when initialResults isn't provided

Reply:
Tests fail without `if (indexInitialResults)` so to me it's already covered in some ways.